### PR TITLE
fetch: fix bug for history retention and decimal

### PR DESCRIPTION
**Pull request by ZhouXing19**
jira:
- https://cockroachlabs.atlassian.net/browse/CC-28309
- https://cockroachlabs.atlassian.net/browse/CC-28310

Release note: fix bug for history retention and decimal

### CODE REVIEW

File: fetch/dataexport/dataexport.go [hunk 1]
```diff
@@ -80,6 +81,10 @@ func scanWithRowIterator(
 	table rowiterator.ScanTable,
 ) error {
 	cw := moltcsv.NewWriter(writer)
+	_, ok := c.(*dbconn.PGConn)
+	if ok {
+		pgxdecimal.Register(c.TypeMap())
+	}
```
Comment (Jeremyyang920) [inline fetch/dataexport/dataexport.go:87]: Do we need to register the type map twice? Since we register it pg.go as well.

Comment (ZhouXing19) [reply fetch/dataexport/dataexport.go:87]: Removed the one in NewPgSource. We have to declare it in the most immediate usage.

**Review by Jeremyyang920 — APPROVED** [review]
Another rabbit hole but LGTM

**Comment by ZhouXing19** [comment]
> Another rabbit hole but LGTM

🐰

**Pull request merged**